src/metabase/api/search.clj

```diff
@@ -309,47 +306,47 @@
   [model search-ctx :- SearchContext]
   (-> (base-query-for-model model search-ctx)
-      (hh/left-join [Table :table] [:= :segment.table_id :table.id])))
+      (sql.helpers/left-join [:metabase_table :table] [:= :segment.table_id :table.id])))
 
 (s/defmethod search-query-for-model "table"
   [model {:keys [current-user-perms table-db-id], :as search-ctx} :- SearchContext]
   (when (seq current-user-perms)
     (let [base-query (base-query-for-model model search-ctx)]
       (add-table-db-id-clause
-        (if (contains? current-user-perms "/")
-          base-query
-          (let [data-perms (filter #(re-find #"^/db/*" %) current-user-perms)]
-            {:select (:select base-query)
-             :from   [[(merge
-                         base-query
-                         {:select [:id :schema :db_id :name :description :display_name :updated_at :initial_sync_status
-                                   [(hx/concat (hx/literal "/db/")
-                                               :db_id
-                                               (hx/literal "/schema/")
-                                               (hsql/call :case
-                                                          [:not= :schema nil] :schema
-                                                          :else               (hx/literal ""))
-                                               (hx/literal "/table/") :id
-                                               (hx/literal "/read/"))
-                                    :path]]})
-                       :table]]
-             :where  (if (seq data-perms)
-                       (into [:or] (for [path data-perms]
-                                     [:like :path (str path "%")]))
-                       [:= 0 1])}))
-        table-db-id))))
+       (if (contains? current-user-perms "/")
+         base-query
+         (let [data-perms (filter #(re-find #"^/db/*" %) current-user-perms)]
+           {:select (:select base-query)
+            :from   [[(merge
+                       base-query
+                       {:select [:id :schema :db_id :name :description :display_name :updated_at :initial_sync_status
+                                 [(h2x/concat (h2x/literal "/db/")
+                                             :db_id
+                                             (h2x/literal "/schema/")
+                                             [:case
+                                              [:not= :schema nil] :schema
+                                              :else               (h2x/literal "")]
+                                             (h2x/literal "/table/") :id
+                                             (h2x/literal "/read/"))
+                                  :path]]})
+                      :table]]
+            :where  (if (seq data-perms)
+                      (into [:or] (for [path data-perms]
+                                    [:like :path (str path "%")]))
+                      [:= 0 1])}))
```

Comment:
This is the same idea as `api.collection/always-false-hsql-expr`; idk if it's worth pulling it out to `h2x` or something to formalize the pattern. It's certainly not a big deal to just write `[:= 0 1]` though.